js/modules/k6/http/request_test.go

```diff
@@ -354,6 +354,22 @@ func TestRequestAndBatch(t *testing.T) {
 			`))
 			assert.NoError(t, err)
 		})
+		t.Run("custom compression", func(t *testing.T) {
+			// We should not try to decode it
+			tb.Mux.HandleFunc("/customcompression", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Encoding", "custom")
+				_, err := w.Write([]byte(`{"custom": true}`))
+				assert.NoError(t, err)
+			}))
+
+			_, err := common.RunString(rt, sr(`
+				let res = http.get("HTTPBIN_URL/customcompression");
+				if (res.json()["custom"] != true) {
+					throw new Error("unexpected body data: " + res.body)
+				}
+			`))
+			assert.NoError(t, err)
```

Comment:
Hmm I wonder if we should be smarter about this and not just ignore the header if we don't support the encoding. Maybe the correct behavior would be to send `Accept-Encoding` and throw an error if the server returns something else?

Reply:
The whole point of this is that as long as we don't actually touch the response, the user can use whatever custom encoding/decoding they like. We have no mechanism to give that any other way, so I think this is better ... although it is possible that I am wrong